flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCOutputFormat.java

```diff
@@ -111,110 +134,117 @@ public void writeRecord(Row row) throws IOException {
 		if (typesArray != null && typesArray.length > 0 && typesArray.length != row.getArity()) {
 			LOG.warn("Column SQL types array doesn't match arity of passed Row! Check the passed array...");
 		}
-		try {
+		synchronized (this) {
```

Comment:
Why do we need synchronized this?

Reply:
we need synchronization because two ```flush()``` may be called by two separated thread. As it is explained in **the purpose of the change**.